subproject1/code/chi-square.R

```diff
@@ -1,23 +1,47 @@
 pacman::p_load(
   ggplot2
-)# This code can check whether you have the desired package and load the package after installing.
+) # This code can check whether you have the desired package and load the package after installing.
 
 here::i_am(
   "subproject1/code/chi-square.R"
 )
 
+data_clean <- readRDS(
+  here::here("data/data_clean.rds") 
+) # read the cleaned data
+
+data_ctg <- subset(data_clean, 
+                   select = c(PNEUMONIA, COPD, DIABETES, CARDIOVASCULAR, RENAL_CHRONIC, ICU)
+                   ) # keep only the wanted categorical variables
+
+sum_table <- data.frame(matrix(ncol = 4, nrow = 0)) # empty dataframe to store Chi-square test outputs
+
+for (i in 1:(ncol(data_ctg)-1)) {
+  chisq_result <- chisq.test(data_ctg$ICU, data_ctg[[i]])
+  output <- c(names(data_ctg)[i], chisq_result$statistic, chisq_result$parameter, chisq_result$p.value)
+  sum_table <- rbind(sum_table, output)
+} # for loop that conducts Chi-square tests 
+
+colnames(sum_table) <- c("Variable", "Chi-square", "df", "p-value") # add dataframe column names
 
 saveRDS(
-  table,
+  sum_table,
   file = here::here("subproject1/output/chi-square/table.rds")
-)
+) # save Chi-square test output table to designated location
+
+for (i in 1:(ncol(data_ctg)-1)) {
+  barchart <- ggplot(data_ctg, aes(x = data_ctg[[i]], fill = ICU)) + 
+    geom_bar( position = "dodge") +
```

Comment:
# save Chi-square test output table to designated location
for (i in 1:(ncol(data_ctg)-1)) {
  variable_name <- names(data_ctg)[i]
  barchart <- ggplot(data_ctg, aes_string(x = variable_name, fill = 'ICU')) + 
    geom_bar(position = "dodge") +
    geom_text(stat='count', aes(label=..count..), vjust=-0.5, position=position_dodge(width=0.9)) +
    xlab(variable_name) +
    ylab("Count") +
    labs(title = paste0("Bar Chart for ", variable_name))+
    theme(plot.title = element_text(hjust = 0.5))
  
  barchart_name <- paste0("barchart_", variable_name, ".png")
  ggsave(
    plot = barchart,
    device = "png",
    filename = file.path(here::here("subproject1/output/chi-square", barchart_name)),
    width = 10,
    height = 6
  )
}